test/unittests/test_second_order_lowpass.cpp

```diff
@@ -0,0 +1,39 @@
+#include <gtest/gtest.h>
+
+#include <cmath>
+#include <path_tracking_pid/details/second_order_lowpass.hpp>
+#include <vector>
+
+using path_tracking_pid::details::SecondOrderLowpass;
+
+constexpr double eps = 1e-6;
+
+TEST(SecondOrderLowpass, StepResponse)
+{
+  double dt = 0.1;
+  double cutoff = 1 / dt / 4;
+  double damping = sqrt(2);
+
+  SecondOrderLowpass filter(cutoff, damping);
+
+  std::vector<double> expected_response = {0.16071,  0.514214, 0.770813, 0.877725, 0.939488,
+                                           0.968659, 0.984211, 0.991911, 0.995898, 0.997907};
+  for (int i = 0; i < static_cast<int>(expected_response.size()); ++i) {
+    SCOPED_TRACE(i);
+
+    auto result = filter.filter(1, dt);
+    EXPECT_NEAR(result, expected_response[i], eps);
+  }
+}
+
```

Comment:
Consider adding tests with different cutoff and/or damping values. And using different values for dt.

Reply:
I've added tests for different damping and cutoff